### PR TITLE
[Misc] [botocore] Add config for botocore => `distributed_tracing` 

### PR DESCRIPTION
**Pull request by Czechh**
## Description

Adds `distributed_tracing` config option for `botocore`. It's defaulted to `True`

Inspired by https://github.com/DataDog/dd-trace-py/pull/1194/files

## Checklist
- [x] Entry added to release notes using [reno](https://docs.openstack.org/reno/latest/user/usage.html)
- [X] Tests provided; and/or
- [ ] Description of manual testing performed and explanation is included in the code and/or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

File: releasenotes/notes/botocore-distributed-tracing-config-cd37b439d6d77dc6.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+features:
+  - |
+    botocore: `distributed_tracing` config: defaults to True
```
Comment (Kyle-Verhoog) [inline releasenotes/notes/botocore-distributed-tracing-config-cd37b439d6d77dc6.yaml:4]:
```suggestion
    botocore: added `distributed_tracing` configuration setting which is enabled by default.
```

File: ddtrace/contrib/botocore/__init__.py [hunk 2]
```diff
@@ -2,19 +2,34 @@
...
-    botocore.session.get_session()
-    lambda_client = session.create_client('lambda', region_name='us-east-1')
-    # Example of instrumented query
-    lambda_client.list_functions()
+Configuration
+~~~~~~~~~~~~~
+
+.. py:data:: ddtrace.config.botocore['distributed_tracing']
+
+   Whether to inject distributed tracing data to requests in SQS and Lambda.
```
Comment (Kyle-Verhoog) [inline ddtrace/contrib/botocore/__init__.py:21]: Let's also mention the environment variable that can be used (`DD_BOTOCORE_DISTRIBUTED_TRACING`)

**Review by Kyle-Verhoog — COMMENTED** [review]
Just a couple minor things otherwise looks great! Thanks for the tests and the documentation enhancement 😄

File: ddtrace/contrib/botocore/__init__.py (hunk 2 above)
Comment (Czechh) [reply ddtrace/contrib/botocore/__init__.py:21]: I added `DD_TRACE_BOTOCORE_DISTRIBUTED_TRACING`  instead of `DD_BOTOCORE_DISTRIBUTED_TRACING`.

Comment (Czechh) [reply ddtrace/contrib/botocore/__init__.py:21]: Actually, after reading the code, I think you were right. I would need to change the one here too. https://github.com/DataDog/dd-trace-py/blob/master/ddtrace/contrib/starlette/__init__.py#L35

Comment (Kyle-Verhoog) [reply ddtrace/contrib/botocore/__init__.py:21]: ah good catch!

File: ddtrace/contrib/starlette/__init__.py [hunk 3]
```diff
@@ -32,7 +32,7 @@
 
    Whether to parse distributed tracing headers from requests received by your Starlette app.
 
-   Can also be enabled with the ``DD_TRACE_STARLETTE_DISTRIBUTED_TRACING`` environment variable.
+   Can also be enabled with the ``DD_STARLETTE_DISTRIBUTED_TRACING`` environment variable.
```
Comment (Kyle-Verhoog) [inline ddtrace/contrib/starlette/__init__.py:35]: Just need to take this change out 😄

**Review by Kyle-Verhoog — APPROVED** [review]
👏 👏

**Pull request merged**